packages/js/product-editor/src/components/variations-table/variations-table.tsx

```diff
@@ -196,22 +159,40 @@ export const VariationsTable = forwardRef<
 	const { createSuccessNotice, createErrorNotice } =
 		useDispatch( 'core/notices' );
 
-	if ( latestVariations && latestVariations !== lastVariations.current ) {
-		lastVariations.current = latestVariations;
-	}
+	const {
+		isLoading,
+		variations,
+		totalCount,
+		onPageChange,
+		onPerPageChange,
+		onFilter,
+		getFilters,
+		hasFilters,
+		clearFilters,
+
+		selected,
+		isSelectingAll,
+		selectedCount,
+		areAllSelected,
+		areSomeSelected,
+		isSelected,
+		onSelect,
+		onSelectPage,
+		onSelectAll,
+		onClearSelection,
 
-	if ( isLoading && lastVariations.current === null ) {
-		return (
-			<div className="woocommerce-product-variations__loading">
-				<Spinner />
-				{ isGeneratingVariations && (
-					<span>
-						{ __( 'Generating variations…', 'woocommerce' ) }
-					</span>
-				) }
-			</div>
-		);
-	}
+		isUpdating,
+		onUpdate,
+		onDelete,
+		onBatchUpdate,
+		onBatchDelete,
+	} = useVariations( { productId } );
+
+	useEffect( () => {
+		if ( isGeneratingVariations ) {
+			clearFilters();
```

Comment:
We should also clear the total items and some of the pagination info, as this was not updating for me.
Also the selected total count didn't update ( I believe ).